### PR TITLE
PIN-3767: Verify empty string for claims

### PR DESCRIPTION
**Pull request by galales**

### CODE REVIEW

File: client-assertion-validation/src/main/scala/it/pagopa/interop/clientassertionvalidation/ClientAssertionValidator.scala [hunk 1]
```diff
@@ -110,8 +110,9 @@ final class NimbusClientAssertionValidator(expectedAudience: Set[String]) extend
 
   private def getOrFail[E, T](value: => T, error: => E): ValidatedNel[E, T] =
     Try(Option(value)) match {
-      case Failure(_) | Success(None) => error.invalidNel
-      case Success(Some(v))           => v.validNel
+      case Failure(_) | Success(None)                 => error.invalidNel
+      case Success(v) if v.exists(_.toString.isEmpty) => error.invalidNel
```
Comment (TonioGela) [inline client-assertion-validation/src/main/scala/it/pagopa/interop/clientassertionvalidation/ClientAssertionValidator.scala]: just bikeshedding
```suggestion
      case Success(v) if v.exists(_.toString.isBlank) => error.invalidNel
```

File: client-assertion-validation/src/test/scala/it/pagopa/interop/clientassertionvalidation/ClientAssertionValidationSpec.scala [hunk 2]
```diff
@@ -91,6 +91,14 @@ class ClientAssertionValidationSpec extends AnyWordSpecLike {
       ) shouldBe Left(NonEmptyList.one(InvalidPurposeIdFormat("not-an-uuid")))
     }
 
+    "fail when purpose id is ampty string" in {
```
Comment (TonioGela) [inline client-assertion-validation/src/test/scala/it/pagopa/interop/clientassertionvalidation/ClientAssertionValidationSpec.scala]:
```suggestion
    "fail when purpose id is empty string" in {
```

File: client-assertion-validation/src/test/scala/it/pagopa/interop/clientassertionvalidation/ClientAssertionValidationSpec.scala [hunk 3]
```diff
@@ -91,6 +91,14 @@ class ClientAssertionValidationSpec extends AnyWordSpecLike {
...
+      val assertion = fastClientAssertionJWT(customClaims = Map(PURPOSE_ID_CLAIM -> ""))
+
+      validateClientAssertion(Some(clientId.toString), assertion, clientAssertionType, grantType)(
+        jwtValidator
+      ) shouldBe Left(NonEmptyList.one(InvalidPurposeIdFormat("")))
```
Comment (TonioGela) [inline client-assertion-validation/src/test/scala/it/pagopa/interop/clientassertionvalidation/ClientAssertionValidationSpec.scala:99]: Testing what I suggested a few lines before
```suggestion
      ) shouldBe Left(NonEmptyList.one(InvalidPurposeIdFormat("   ")))
```

**Review by TonioGela — APPROVED** [review]

File: client-assertion-validation/src/test/scala/it/pagopa/interop/clientassertionvalidation/ClientAssertionValidationSpec.scala [hunk 4]
```diff
@@ -91,6 +91,14 @@ class ClientAssertionValidationSpec extends AnyWordSpecLike {
       ) shouldBe Left(NonEmptyList.one(InvalidPurposeIdFormat("not-an-uuid")))
     }
 
+    "fail when purpose id is ampty string" in {
+      val assertion = fastClientAssertionJWT(customClaims = Map(PURPOSE_ID_CLAIM -> ""))
```
Comment (TonioGela) [inline client-assertion-validation/src/test/scala/it/pagopa/interop/clientassertionvalidation/ClientAssertionValidationSpec.scala:95]: To test what I suggested
```suggestion
      val assertion = fastClientAssertionJWT(customClaims = Map(PURPOSE_ID_CLAIM -> " 
  "))
```

File: client-assertion-validation/src/main/scala/it/pagopa/interop/clientassertionvalidation/ClientAssertionValidator.scala (hunk 1 above)
Comment (galales) [reply client-assertion-validation/src/main/scala/it/pagopa/interop/clientassertionvalidation/ClientAssertionValidator.scala]: nice, thanks!

**Pull request merged**